src/Sulu/Bundle/EventLogBundle/Infrastructure/Doctrine/Repository/EventRecordRepository.php

```diff
@@ -62,13 +68,51 @@ public function createForDomainEvent(DomainEvent $domainEvent): EventRecordInter
         return $eventRecord;
     }
 
-    public function add(EventRecordInterface $eventRecord): void
+    protected function getClassName(): string
+    {
+        return $this->entityRepository->getClassName();
+    }
+
+    protected function getQueryBuilder(): QueryBuilder
```

Comment:
would call this `getInsertQueryBuilder` and return a query builder that is ready to be executed 🙂